File: src/main/java/me/ksviety/teleporter/cache/TeleportationCache.java
package me.ksviety.teleporter.cache;

import me.ksviety.teleporter.Cache;

import java.util.Collection;
import java.util.Set;

public abstract class TeleportationCache implements Cache {
    private final Set<String> alreadyTeleportedPlayers;

    protected TeleportationCache(Set<String> alreadyTeleportedPlayers) {
        this.alreadyTeleportedPlayers = alreadyTeleportedPlayers;
    }

    public Collection<String> getAlreadyTeleportedPlayers() {
        return alreadyTeleportedPlayers;
    }

    public void addPlayerAsTeleported(String name) {
        alreadyTeleportedPlayers.add(name);
    }
}

File: src/main/kotlin/me/ksviety/teleporter/cache/TeleportationCache.kt
package me.ksviety.teleporter.cache

class TeleportationCache(alreadyTeleportedPlayers: Set<String>) {
    private val _alreadyTeleportedPlayers = HashSet(alreadyTeleportedPlayers)

    val alreadyTeleportedPlayers: Collection<String>
        get() {
            return _alreadyTeleportedPlayers
        }

    fun addPlayerAsTeleported(name: String) {
        _alreadyTeleportedPlayers.add(name)
    }
}
